src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.NoSuchElementException;
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
+        countryDao = new CountryDaoImpl(sessionFactory);
+    }
+
     @Override
     public Country add(Country country) {
-        return null;
+        countryDao.add(country);
+        return country;
```

Comment:
You should handle the situation if `countryDao.add(country)` throws an exception. Consider wrapping this in a try-catch block to handle any potential `DataProcessingException` that might be thrown from the DAO layer.